lib/services/livesync/playground/preview-app-livesync-service.ts

```diff
@@ -0,0 +1,136 @@
+		});
+		await this.$playgroundQrCodeGenerator.generateQrCodeForCurrentApp();
+	}
+
+	public async syncFiles(data: IPreviewAppLiveSyncData, files: string[]): Promise<void> {
+		for (const device of this.$previewSdkService.connectedDevices) {
+			await this.syncFilesOnDeviceSafe(data, device, files);
+		}
+	}
+
+	public async stopLiveSync(): Promise<void> {
+		this.$previewSdkService.removeAllListeners();
+		this.$previewSdkService.stop();
+	}
+
+	private async syncFilesOnDeviceSafe(data: IPreviewAppLiveSyncData, device: Device, files?: string[]): Promise<void> {
+		await this.$previewAppPluginsService.comparePluginsOnDevice(device);
+		this.showWarningsForNativeFiles(files);
+
+		this.$logger.info(`Start syncing changes on device ${device.id}.`);
+
+		try {
+			await this.syncFilesOnDevice(data, device, files);
+			this.$logger.info(`Successfully synced changes on device ${device.id}.`);
+		} catch (err) {
+			this.$logger.warn(`Unable to apply changes on device ${device.id}. Error is: ${JSON.stringify(err, null, 2)}.`);
+		}
+	}
+
+	private async syncFilesOnDevice(data: IPreviewAppLiveSyncData, device: Device, files?: string[]): Promise<void> {
+		const { appFilesUpdaterOptions, env, projectDir } = data;
+		const platform = device.platform;
+		const projectData = this.$projectDataService.getProjectData(projectDir);
+		const platformData = this.$platformsData.getPlatformData(platform, projectData);
+
+		await this.preparePlatform(platform, appFilesUpdaterOptions, env, projectData);
+
+		const payloads = this.getFilePayloads(platformData, projectData, files);
+		await this.$previewSdkService.applyChanges(payloads);
+	}
+
+	private getFilePayloads(platformData: IPlatformData, projectData: IProjectData, files?: string[]): FilePayload[] {
+		const appFolderPath = path.join(projectData.projectDir, APP_FOLDER_NAME);
+		const platformsAppFolderPath = path.join(platformData.appDestinationDirectoryPath, APP_FOLDER_NAME);
+
+		if (files) {
+			files = files.map(file => path.join(platformsAppFolderPath, path.relative(appFolderPath, file)));
+		} else {
+			files = this.$projectFilesManager.getProjectFiles(platformsAppFolderPath);
+		}
+
+		const filesToTransfer = files
+			.filter(file => file.indexOf(TNS_MODULES_FOLDER_NAME) === -1)
+			.filter(file => file.indexOf(APP_RESOURCES_FOLDER_NAME) === -1)
+			.filter(file => !_.includes(this.excludedFiles, path.basename(file)))
+			.filter(file => !_.includes(this.excludedFileExtensions, path.extname(file)));
+
+		this.$logger.trace(`Transferring ${filesToTransfer.join("\n")}.`);
+
+		const payloads = filesToTransfer
+			.map(file => {
+				const filePayload: FilePayload = {
+					event: PreviewSdkEventNames.CHANGE_EVENT_NAME,
+					file: path.relative(platformsAppFolderPath, file),
+					binary: isTextOrBinary.isBinarySync(file),
+					fileContents: ""
+				};
+
+				if (filePayload.binary) {
+					const bitmap =  <string>this.$fs.readFile(file);
+					const base64 = new Buffer(bitmap).toString('base64');
```

Comment:
`new Buffer` is deprecated, we should use Buffer.from or maybe `this.$fs.readFile(file, { encoding: "base64" })`